code/processes/gateway.q

```diff
@@ -446,7 +450,7 @@ asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
    :();
    ];
   ];
- query:({[u;q]$[`.pm.execas ~ key `.pm.execas;value (`.pm.execas; q; u);value q]}; .z.u; query);
+ query:({[u;q]$[`.pm.execas ~ key `.pm.execas;value (`.pm.execas; q; u);`.pm.valp ~ key `.pm.valp; .pm.valp q; @[value;q;{.lg.e[`gw;"Failed to execute query."]}]]}; .z.u; query;.gw.valp);
```

Comment:
I don't think this will work as you are defining the function with 2 args but passing in 3 (and same for the "query" function definition later)
Also I don't think you want to have an error trap in this remote execution